include/seastar/http/request.hh

```diff
@@ -121,6 +122,25 @@ struct request {
         return res->second;
     }
 
+    /**
+     * Search for the last path parameter of a given key
+     * @param key the path paramerter key
+     * @return the unescaped path parameter value, if it exists and can be path decoded successfully, otherwise it
+     *  returns an empty string
+     */
+    sstring get_path_param(const sstring& key) const {
+        if (!param.exists(key)) {
+            return "";
+        }
+        auto& raw_path_param = param.at(key);
+        auto decoded_path_param = sstring{};
```

Comment:
This does the lookup twice:
```suggestion
        auto res = param.find(key);
        if (res == param.end()) {
            return "";
        }
        auto& raw_path_param = res->second;
        sstring decoded_path_param;

Reply:
Nice, yeah I just had to expose these methods